src/main/java/com/yurjinia/auth/service/AuthService.java

```diff
@@ -134,6 +142,27 @@ private RegistrationRequest payloadOAuthUser(String email, String firstname, Str
                 .build();
     }
 
+    public void logout(HttpServletRequest request, HttpServletResponse response) {
+        String requestHeader = request.getHeader(JwtConstants.AUTHORIZATION_HEADER);
+
+        if (StringUtils.hasText(requestHeader) && requestHeader.startsWith(JwtConstants.TOKEN_PREFIX)) {
+            String token = requestHeader.substring(JwtConstants.TOKEN_PREFIX.length());
+            long expirationTime = jwtService.getExpirationTime(token);
+            var auth = SecurityContextHolder.getContext().getAuthentication();
+
+            if (auth != null) {
+                SecurityContextHolder.getContext().setAuthentication(null);
+
+                if (jwtBlacklistService.isTokenBlacklisted(token)) {
```

Comment:
I think it's better to validate if token blacklisted at the beginning, because in this change you're checking if token blacklisted, after you're doing `.setAuthentication(null)`